pylint/checkers/variables.py

```diff
@@ -703,17 +703,18 @@ def _uncertain_nodes_in_except_blocks(
         for other_node in found_nodes:
             other_node_statement = other_node.statement(future=True)
             # Only testing for statements in the except block of TryExcept
-            if not (
-                isinstance(other_node_statement.parent, nodes.ExceptHandler)
-                and isinstance(other_node_statement.parent.parent, nodes.TryExcept)
-            ):
+            closest_except_handler = utils.get_node_first_ancestor_of_type(
+                other_node_statement, nodes.ExceptHandler
+            )
+            if not closest_except_handler:
                 continue
             # If the other node is in the same scope as this node, assume it executes
-            if other_node_statement.parent.parent_of(node):
+            if closest_except_handler.parent_of(node):
                 continue
+            closest_try_except: nodes.TryExcept = closest_except_handler.parent
```

Comment:
I think it would be better to add an ignore wherever that is necessary and then wait for `astroid`'s typing to be updated. If we update `parent` of `ExceptHandler` at some point we wouldn't notice that this typing is redundant.

Reply:
`mypy` didn't make me do this, I just thought I was helping readability! I'll change this to an assert instead to make more explicit that I'm depending on the parent being a TryExcept.

Reply:
Ah okay, never mind then. I'm not sure about performance impact but typing is probably better than asserts? So we should probably revert 😅